### ISSUE TITLE
Keep receiving "mReversing is false. Don't call initChildren" in Android 11.0.1

### ISSUE THREAD

**Issue by cooltey** [maplibre/maplibre-native#2586]
**Describe the bug**
We have implemented the `org.maplibre.gl:android-sdk:11.0.1` in [our app](https://github.com/wikimedia/apps-android-wikipedia). When launching the map, we can observe from the Logcat that keeps outputting "mReversing is false. Don't call initChildren.".

**To Reproduce**
1. Build the `devDebug` variant of the app
2. Go to "Places" in the More menu.

**Expected behavior**
Do not consistently output the "mReversing is false. Don't call initChildren" message.

**Screenshots**
```
2024-07-09 16:25:23.458 15810-15810 AnimatorSet             org.wikipedia.dev                    D  mReversing is false. Don't call initChildren.
2024-07-09 16:25:23.670 15810-15810 AnimatorSet             org.wikipedia.dev                    D  mReversing is false. Don't call initChildren.
2024-07-09 16:25:23.670 15810-15810 AnimatorSet             org.wikipedia.dev                    D  mReversing is false. Don't call initChildren.
2024-07-09 16:25:23.958 15810-15810 AnimatorSet             org.wikipedia.dev                    D  mReversing is false. Don't call initChildren.
2024-07-09 16:25:24.461 15810-15810 AnimatorSet             org.wikipedia.dev                    D  mReversing is false. Don't call initChildren.
2024-07-09 16:25:25.057 15810-15810 AnimatorSet             org.wikipedia.dev                    D  mReversing is false. Don't call initChildren.
2024-07-09 16:25:25.556 15810-15810 AnimatorSet             org.wikipedia.dev                    D  mReversing is false. Don't call initChildren.
```

**Platform information (please complete the following information):**
- OS: One UI 6
- Platform: android
- Version 14
- MapLibre native version: 11.0.1

**Comment by louwers** [comment]
I didn't know the Wikipedia app used MapLibre Native (or is about to), that's neat!

For reference the source code of `AnimatorSet` https://android.googlesource.com/platform/prebuilts/fullsdk/sources/android-30/+/refs/heads/androidx-print-release/android/animation/AnimatorSet.java

I am happy to try to build your app, but if you could provide some more details how you are using this library, that would be helpful.

**Comment by louwers** [comment]
@cooltey Unable to reproduce this on Samsung S10, Android 12.

**Comment by cooltey** [comment]
> I didn't know the Wikipedia app used MapLibre Native (or is about to), that's neat!
> 
> For reference the source code of `AnimatorSet` https://android.googlesource.com/platform/prebuilts/fullsdk/sources/android-30/+/refs/heads/androidx-print-release/android/animation/AnimatorSet.java
> 
> I am happy to try to build your app, but if you could provide some more details how you are using this library, that would be helpful. Can you pin pint which MapLibre Native methods you call?

Hi @louwers! 

Thanks for taking the time to look into this issue, I really appreciate it. The Wikipedia app uses the library to show the map with custom markers and mapstyle. 

For the XML, we use `<org.maplibre.android.maps.MapView>` in a `FrameLayout`, and some other view elements in the `FrameLayout` on top of the `MapView`.

In the code, we apply theme-based mapstyle JSON files, set some custom icons to the markers and compass, and slightly shift the gravity and margins of those icons.


> @cooltey Unable to reproduce this on Samsung S10, Android 12.

Did you build the `devDebug` variant of the app? I can reproduce it on both of my Android devices (Android 14).

Thanks!

**Comment by louwers** [comment]
@cooltey Yes I tried `devDebug`... Could you try using the sample app in this repository (`platform/android`) to see if it occurs there?

**Comment by cooltey** [comment]
> @cooltey Yes I tried `devDebug`... Could you try using the sample app in this repository (`platform/android`) to see if it occurs there?

Sorry, I am having difficulty building a sample app in the `platform/android`, it keeps giving me the error message about the `CMake Error`

I tried this on my Windows and Mac and got the same error messages. 

```
CMake Error at C:[PATH]maplibre-native/vendor/mapbox-base.cmake:88 (set_target_properties):
  set_target_properties Can not find target to add properties to:
  mapbox-base-variant
Call Stack (most recent call first):
 C:[PATH]maplibre-native/CMakeLists.txt:1341 (include)
```

**Comment by louwers** [comment]
@cooltey You need to check out all the submodules.

```
git submodule foreach git reset --hard
git submodule update --init --recursive
```

**Comment by cooltey** [comment]
> @cooltey You need to check out all the submodules.
> 
> ```
> git submodule foreach git reset --hard
> git submodule update --init --recursive
> ```

Thanks! It works!

Just checked the Debug Mode in the sample app, when clicking on the "layer" icon on the right-bottom, `mReversing is false. Don't call initChildren.` will show up two times (but not consistently). 

Our app does [some custom configurations](https://github.com/wikimedia/apps-android-wikipedia/blob/3b0c6397c3e05ec217684f12a5d75e3603f11ae6/app/src/main/java/org/wikipedia/places/PlacesFragment.kt#L469-L498) for `ClusterOptions` and add layers to the style. Will check if it is the root cause of this issue.
